### DISCUSSION TITLE
Serverpod docs outdated?

### DISCUSSION THREAD

**Discussion by yahu1031** [serverpod/serverpod#167] [Q&A] [answered]
I was trying to make authentications following the docs. But I found that we don't have some variables/properties in `Client`

```dart
class Client extends ServerpodClient {
  late final _EndpointExample example;

  Client(String host,
      {SecurityContext? context,
      ServerpodClientErrorCallback? errorHandler,
      AuthenticationKeyManager? authenticationKeyManager})
      : super(host, Protocol.instance,
            context: context,
            errorHandler: errorHandler,
            authenticationKeyManager: authenticationKeyManager) {
    example = _EndpointExample(this);
  }

  @override
  Map<String, EndpointRef> get endpointRefLookup => {
        'example': example,
      };

  @override
  Map<String, ModuleEndpointCaller> get moduleLookup => {};
}
```

This is what we have in template-generated code with version `v0.9.7`.

But the docs with the same version use `client.modules` to get the caller in the `Sessions` class [here](https://docs.serverpod.dev/concepts/authentication#add-sign-in-buttons-in-your-code).

**Comment by vlidholt** [comment, marked as answer]
The `modules` field isn't generated if there are no modules added to the project. There is some information here about adding modules: https://docs.serverpod.dev/concepts/modules
